pkg/client/client.go

```diff
@@ -320,13 +320,18 @@ type ProviderUpdateSummary struct {
 func (c *Client) CheckForProviderUpdates(ctx context.Context) ([]ProviderUpdateSummary, error) {
 	var summary []ProviderUpdateSummary
 	for _, p := range c.Providers {
+		// if version is latest it means there is no update as DownloadProvider will download the latest version automatically
+		if strings.Compare(p.Version, "latest") == 0 {
+			c.Logger.Debug("version is latest already", "provider", p.Name, "version", p.Version)
+			continue
+		}
 		version, err := c.Hub.CheckProviderUpdate(ctx, p)
 		if err != nil {
-			c.Logger.Warn("Failed check provider update", "provider", p.Name)
+			c.Logger.Warn("Failed check provider update", "provider", p.Name, "error", err)
 			continue
 		}
-
 		if version == nil {
+			c.Logger.Debug("version is latest already", "provider", p.Name, "version", p.Version)
```

Comment:
```suggestion
			c.Logger.Debug("already at latest version", "provider", p.Name, "version", p.Version)
```

Because otherwise it would conflict with the new message at line 325, and not really clear either